src/utils/utils_posix_common.c

```diff
@@ -38,3 +48,57 @@ int utils_gettid(void) {
     return syscall(SYS_gettid);
 #endif
 }
+
+int utils_close_fd(int fd) { return close(fd); }
+
+static umf_result_t errno_to_umf_result(int err) {
```

Comment:
Unused on MacOSX:
https://github.com/oneapi-src/unified-memory-framework/actions/runs/9710408763/job/26801228405?pr=558
```
/Users/runner/work/unified-memory-framework/unified-memory-framework/src/utils/utils_posix_common.c:54:21: error: unused function 'errno_to_umf_result' [-Werror,-Wunused-function]
[ 28%] Building C object src/CMakeFiles/umf.dir/utils/utils_posix_math.c.o
static umf_result_t errno_to_umf_result(int err) {
                    ^
1 error generated.
make[2]: *** [src/CMakeFiles/umf.dir/utils/utils_posix_common.c.o] Error 1
```

Reply:
yeah, I put this function under `#ifndef __APPLE__`